spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/support/AbstractRibbonCommand.java

```diff
@@ -93,16 +93,31 @@ protected AbstractRibbonCommand(Setter setter, LBC client,
 				0).get();
 		int commandHystrixTimeout = dynamicPropertyFactory.getIntProperty("hystrix.command." + commandKey + ".execution.isolation.thread.timeoutInMilliseconds",
 				0).get();
-		int ribbonReadTimeout = config == null ? RibbonClientConfiguration.DEFAULT_READ_TIMEOUT :
-				config.get(IClientConfigKey.Keys.ReadTimeout, RibbonClientConfiguration.DEFAULT_READ_TIMEOUT).intValue();
-		int ribbonConnectTimeout = config == null ? RibbonClientConfiguration.DEFAULT_CONNECT_TIMEOUT :
-				config.get(IClientConfigKey.Keys.ConnectTimeout, RibbonClientConfiguration.DEFAULT_CONNECT_TIMEOUT).intValue();
-		int ribbonTimeout = ribbonConnectTimeout + ribbonReadTimeout;
+		int ribbonTimeout;
+		if (config == null) {
+			ribbonTimeout = RibbonClientConfiguration.DEFAULT_READ_TIMEOUT + RibbonClientConfiguration.DEFAULT_CONNECT_TIMEOUT;
+		} else {
+			int ribbonReadTimeout = dynamicPropertyFactory.getIntProperty(
+				commandKey + "." + config.getNameSpace() + ".ReadTimeout",
+				config.get(IClientConfigKey.Keys.ReadTimeout, RibbonClientConfiguration.DEFAULT_READ_TIMEOUT)).get();
+			int ribbonConnectTimeout = dynamicPropertyFactory.getIntProperty(
+				commandKey + "." + config.getNameSpace() + ".ConnectTimeout",
+				config.get(IClientConfigKey.Keys.ConnectTimeout, RibbonClientConfiguration.DEFAULT_CONNECT_TIMEOUT)).get();
+			//Because ribbon.maxAutoRetries defaults to 0, here we set default Value to 0
+			int maxAutoRetries = dynamicPropertyFactory.getIntProperty(
+				commandKey + "." + config.getNameSpace() + ".MaxAutoRetries",
+				config.get(IClientConfigKey.Keys.MaxAutoRetries, 0)).get();
+			//Because ribbon.maxAutoRetriesNextServer defaults to 1, here we set default Value to 1
+			int maxAutoRetriesNextServer = dynamicPropertyFactory.getIntProperty(
+				commandKey + "." + config.getNameSpace() + ".MaxAutoRetriesNextServer",
+				config.get(IClientConfigKey.Keys.MaxAutoRetriesNextServer, 1)).get();
+			ribbonTimeout = (ribbonReadTimeout + ribbonConnectTimeout) * (maxAutoRetries + 1) * (maxAutoRetriesNextServer + 1);
```

Comment:
are you adding `+1` to `maxAutoRetries` and `maxAutoRetriesNextServer` because the first attempt does not account towards the total number of times tried?

Reply:
Yeah!
In the official Ribbon wiki we can see the following introduction : 
```
# Max number of retries on the same server (excluding the first try)
sample-client.ribbon.MaxAutoRetries=1

# Max number of next servers to retry (excluding the first server)
sample-client.ribbon.MaxAutoRetriesNextServer=1
```
By the way, the default of these two parameters is:
```
ribbon.MaxAutoRetries is 0
ribbon.MaxAutoRetriesNextServer is 1
```
So the total Hystrix timeout should include the first and time all retries